src/Lykke.AlgoStore.CSharp.AlgoTemplate.Services/Functions/ADX/DirectionalMovementIndexPlusFunction.cs

```diff
@@ -0,0 +1,167 @@
+﻿using Lykke.AlgoStore.CSharp.Algo.Core.Candles;
+using Lykke.AlgoStore.CSharp.Algo.Core.Functions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Lykke.AlgoStore.CSharp.AlgoTemplate.Services.Functions.ADX
+{
+    public class DirectionalMovementIndexPlusFunction : IFunction
+    {
+        private readonly int _period;
+        private readonly bool _isAverageTrueRangeSet = false;
+        private int _samples { get; set; }
+
```

Comment:
I believe "new DMIParameters();" is ignored during initialization and is not needed here.